test/integ_test/aml_model_test/test_aml_model.py

```diff
@@ -61,7 +49,38 @@ def test_aml_model():
         assert Path(onnx_model.model_path).is_file()
 
 
-def generate_olive_workspace_config(workspace_config_path):
+def test_aml_model_download():
+    pytorch_model = get_pytorch_model()
+    tmp_dir = tempfile.TemporaryDirectory()
+    tmp_dir_path = Path(tmp_dir.name).resolve()
+
+    download_path = pytorch_model.download_model(tmp_dir_path)
+    assert Path(download_path).is_file()
+
+
+def get_pytorch_model():
+    workspace_config = get_olive_workspace_config()
+    azureml_client_config = AzureMLClientConfig(**workspace_config)
+    pytorch_model = PyTorchModel(
+        model_path={
+            "type": "azureml_model",
+            "config": {
+                "azureml_client": azureml_client_config,
+                "name": "bert_glue",
+                "version": 10,
+            },
+        },
+        io_config={
+            "input_names": ["input_ids", "attention_mask", "token_type_ids"],
+            "input_shapes": [[1, 128], [1, 128], [1, 128]],
+            "input_types": ["int64", "int64", "int64"],
+            "output_names": ["output"],
+        },
+    )
+    return pytorch_model
+
+
+def get_olive_workspace_config():
     subscription_id = os.environ.get("WORKSPACE_SUBSCRIPTION_ID")
     if subscription_id is None:
         raise Exception("Please set the environment variable WORKSPACE_SUBSCRIPTION_ID")
```

Comment:
ditto

Reply:
Will address in the future when we refactor the other exceptions in Olive.